pyrit/scenario/core/scenario.py

```diff
@@ -263,6 +219,8 @@ async def initialize_async(
         # Set instance variables from parameters
         self._objective_target = objective_target
         self._objective_target_identifier = objective_target.get_identifier()
+        self._dataset_config_provided = dataset_config is not None
+        self._dataset_config = dataset_config if dataset_config is not None else self.default_dataset_config()
```

Comment:
nit - this code reads the same way right? or do we need to check for None

```suggestion
        self._dataset_config = dataset_config if dataset_config else self.default_dataset_config()
```